readthedocs/projects/tasks.py

```diff
@@ -1005,3 +1007,15 @@ def sync_callback(_, version_pk, commit, *args, **kwargs):
     """
     fileify(version_pk, commit=commit)
     update_search(version_pk, commit=commit)
+
+
+@app.task()
+def finish_inactive_builds():
+    query = (~Q(state=BUILD_STATE_FINISHED) &
+             Q(date__lte=datetime.datetime.now() - datetime.timedelta(minutes=45)))
+    # TODO: consider ``poject.container_time_limit`` since it could be bigger than 45 minutes
+    for build in Build.objects.filter(query):
+        build.success = False
+        build.state = BUILD_STATE_FINISHED
+        build.error = 'This build was terminated due to inactivity.'
```

Comment:
On my PR, I'm accumulating these error messages in one spot, which might make sense. I'm storing these error messages all in a central `projects.exceptions` file. This could be implemented as an exception, but I'm not sure if the exception will be raised otherwise -- we could combine the message for this and the timeout kill though.